### ISSUE TITLE
error while using sklearn's XGBoost wrappers to plot dtreeviz

### ISSUE THREAD

**Issue by gustavomrg** [parrt/dtreeviz#164]
I am training an XGBoost Model such as:

```
## Training XGBoost Model
xgb_params = {
    'objective': 'multi:softmax',
    'reg_lambda': 0.8,
    'reg_alpha': 0.4,
    'max_depth': 5,
    'max_delta_step': 1,
    'random_state': 42,
    'n_jobs': -1
}
clf = xgb.XGBClassifier(**xgb_params)
clf.fit(X, y, eval_metric='auc', verbose=True)
```

and I tried to plot the three such as:

```
ftr_names = X.columns.values.tolist()
viz = dtreeviz(clf1, X, y, feature_names=ftr_names, class_names=[0, 1, 2, 3, 4, 5]) 
viz
```

which returns me:

```
ValueError: You need to pass in a tree_index parameter.
```

So, I added the `tree_index` parameter following #156:

```
ftr_names = X.columns
viz = dtreeviz(clf1, X, y, feature_names=ftr_names, class_names=[0, 1, 2, 3, 4, 5], tree_index=0)
viz
```

which prompts:

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-80-4b2f3e7c1ffc> in <module>
      1 ftr_names = X.columns
----> 2 viz = dtreeviz(clf1, X, y, feature_names=ftr_names, class_names=[0, 1, 2, 3, 4, 5], tree_index=0)
      3 viz

/usr/local/lib/python3.8/site-packages/dtreeviz/trees.py in dtreeviz(tree_model, x_data, y_data, feature_names, target_name, class_names, tree_index, precision, orientation, instance_orientation, show_root_edge_labels, show_node_labels, show_just_path, fancy, histtype, highlight_path, X, max_X_features_LR, max_X_features_TD, depth_range_to_display, label_fontsize, ticks_fontsize, fontname, title, title_fontsize, colors, scale)
    812                                 highlight_node=node.id in highlight_path)
    813             else:
--> 814                 regr_split_viz(node, X_data, y_data,
    815                                filename=f"{tmp}/node{node.id}_{os.getpid()}.svg",
    816                                target_name=shadow_tree.target_name,

/usr/local/lib/python3.8/site-packages/dtreeviz/trees.py in regr_split_viz(node, X_train, y_train, target_name, filename, y_range, ticks_fontsize, label_fontsize, fontname, precision, X, highlight_node, colors)
   1120     ax.tick_params(colors=colors['tick_label'])
   1121 
-> 1122     feature_name = node.feature_name()
   1123 
   1124     ax.set_xlabel(f"{feature_name}", fontsize=label_fontsize, fontname=fontname, color=colors['axis_label'])

/usr/local/lib/python3.8/site-packages/dtreeviz/models/shadow_decision_tree.py in feature_name(self)
    512 
    513         if self.shadow_tree.feature_names is not None:
--> 514             return self.shadow_tree.feature_names[self.feature()]
    515         return None
    516 

/usr/local/lib/python3.8/site-packages/dtreeviz/models/shadow_decision_tree.py in feature(self)
    506         """Returns feature index used at this node"""
    507 
--> 508         return self.shadow_tree.get_node_feature(self.id)
    509 
    510     def feature_name(self) -> (str, None):

/usr/local/lib/python3.8/site-packages/dtreeviz/models/xgb_decision_tree.py in get_node_feature(self, id)
     75         feature_name = self._get_column_value("Feature")[id]
     76         try:
---> 77             return self.feature_names.index(feature_name)
     78         except ValueError as error:
     79             return self.__class__.NO_FEATURE

AttributeError: 'Index' object has no attribute 'index'

```

Am I missing something?

---
Versions:

python==3.8.12
scikit-learn==1.0.1
dtreeviz==1.3.1
xgboost==1.5.0

**Comment by tlapusan** [comment]
hi @gustavomrg, what's the type of ftr_names ?

it should be a list.

**Comment by gustavomrg** [comment]
Hi @tlapusan, thanks for the very fast answer. I made a change so it will become a list:

```
ftr_names = X.columns.values.tolist()
viz = dtreeviz(clf, X, y, feature_names=ftr_names, class_names=["0","1", "2", "3", "4", "5"], tree_index=0)
viz.view()
```

 however, I get the following error:

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-49-55e17e66bce6> in <module>
----> 1 viz.view()

/usr/local/lib/python3.8/site-packages/dtreeviz/trees.py in view(self)
     45 
     46     def view(self):
---> 47         svgfilename = self.save_svg()
     48         view(svgfilename)
     49 

/usr/local/lib/python3.8/site-packages/dtreeviz/trees.py in save_svg(self)
     52         tmp = tempfile.gettempdir()
     53         svgfilename = os.path.join(tmp, f"DTreeViz_{os.getpid()}.svg")
---> 54         self.save(svgfilename)
     55         return svgfilename
     56 

/usr/local/lib/python3.8/site-packages/dtreeviz/trees.py in save(self, filename)
     77         cmd = ["dot", f"-T{format}", "-o", filename, dotfilename]
     78         # print(' '.join(cmd))
---> 79         run(cmd, capture_output=True, check=True, quiet=False)
     80 
     81         if filename.endswith(".svg"):

TypeError: 'module' object is not callable
```

**Comment by parrt** [comment]
I wonder how `run` became a module?

**Comment by tlapusan** [comment]
> I wonder how `run` became a module?

@parrt Indeed, it's strange,  "run" is a function from builtins.py module. 

@gustavomrg could you run the examples from these notebook ?https://github.com/parrt/dtreeviz/blob/master/notebooks/dtreeviz_xgboost_visualisations.ipynb
If the error persist, that means there is some project env issues. Thanks

**Comment by parrt** [comment]
I think this could be related to fix I made recently for 1.3.2 dtreeviz:

```
if graphviz.__version__ <= '0.17':
    graphviz.backend.run(cmd, capture_output=True, check=True, quiet=False)
else:
    graphviz.backend.execute.run_check(cmd, capture_output=True, check=True, quiet=False)
```

Maybe this will take care of the `run()` issue. @gustavomrg I think this will fix it so I will close it.
